vktrace/vktrace_replay/vkreplay_vkreplay.cpp

```diff
@@ -2256,16 +2166,183 @@ VkResult vkReplay::manually_replay_vkAllocateMemory(packet_vkAllocateMemory *pPa
+        // Compare trace file platform to replay platform.
+        // If any of the values are null/zero, it is unknown, and we'll consider the platform to not match
+        platformMatch = (m_replay_endianess == m_pFileHeader->endianess) & (m_replay_ptrsize == m_pFileHeader->ptrsize) &
+                        (m_replay_arch == m_pFileHeader->arch) & (m_replay_os == m_pFileHeader->os) &
+                        (m_replay_gpu == m_pGpuinfo->gpu_id) & (m_replay_drv_vers == m_pGpuinfo->gpu_drv_vers) &
+                        (m_replay_gpu != 0) & (m_replay_drv_vers != 0) & (m_pGpuinfo->gpu_id != 0) &
+                        (m_pGpuinfo->gpu_drv_vers != 0) & (strlen((char *)&m_replay_arch) != 0) &
+                        (strlen((char *)&m_replay_os) != 0) & (strlen((char *)&m_pFileHeader->arch) != 0) &
+                        (strlen((char *)&m_pFileHeader->os) != 0);
+    }
+
+    if (m_pFileHeader->portability_table_valid && platformMatch != 1) {
+        long saveFilePos;
+        size_t amIdx;
+        static size_t amSearchPos = 0;
+
+        // Save current file position so we can restore it
+        saveFilePos = ftell(tracefp);
+
+        // Search for all subsequent calls to BIM/BBM calls that use the memory allocated by this call to vkAM
+
+        // First find this vkAM call in portabilityTable
+        pPacket->header = (vktrace_trace_packet_header *)((PBYTE)pPacket - sizeof(vktrace_trace_packet_header));
+        for (amIdx = amSearchPos; amIdx < portabilityTable.size(); amIdx++) {
+            FSEEK(tracefp, (long)portabilityTable[amIdx], SEEK_SET);
+            FREAD(&packetHeader1, sizeof(vktrace_trace_packet_header), 1, tracefp);  // Read the packet header
+
+            if (packetHeader1.global_packet_index == pPacket->header->global_packet_index &&
+                packetHeader1.packet_id == VKTRACE_TPI_VK_vkAllocateMemory) {
+                // Found it
+                // Save away the vkAM return val from the trace file
+                traceAllocateMemoryRval =
+                    *((VkDeviceMemory *)((PBYTE)pPacket->header + pPacket->header->size - sizeof(VkDeviceMemory *)));
+
+                // Save the index where we will start the next search for vkAM
+                amSearchPos = amIdx + 1;
+                break;
+            }
+        }
+        if (amIdx == portabilityTable.size()) {
+            // Didn't find the current vkAM packet, something is wrong with the trace file.
+            // Just use the index from the trace file and attempt to continue.
+            vktrace_LogError("Replay of vkAllocateMemory() failed, trace file may be corrupt.");
             replayResult = m_vkFuncs.real_vkAllocateMemory(remappedDevice, pPacket->pAllocateInfo, NULL, &local_mem.replayGpuMem);
-        } else {
-            vktrace_LogError("vkAllocateMemory() failed, couldn't find memory type for memoryTypeIndex");
-            return VK_ERROR_VALIDATION_FAILED_EXT;
+            fseek(tracefp, saveFilePos, SEEK_SET);
```

Comment:
Can't use the FSEEK macro because we already logged an error and called real_vkAllocateMemory. If we called the FSEEK macro we would potentially do those again.